File: csharp/HyperSwitchAI/src/ListKeys.cs
using System;
using System.Net.Http;
using System.Text.Json;
using System.Threading.Tasks;
using DotNetEnv;

namespace HyperSwitchAI
{
    public class KeyInfo
    {
        public string? keyId { get; set; }
        public string? provider { get; set; }
        // Note: API key values and encryption fragments are not included in the response
    }

    public class ListKeysResponse
    {
        public KeyInfo[]? keys { get; set; }
    }

    public class ListKeys
    {
        private const string BASE_URL = "https://api.hyperswitchai.com";

        public static async Task RunAsync()
        {
            try
            {
                Env.Load();
                var token = await CacheToken.GetToken();

                // List all API keys
                using (var client = new HttpClient())
                {
                    client.DefaultRequestHeaders.Add("Authorization", $"Bearer {token}");
                    var response = await client.GetAsync($"{BASE_URL}/admin/keys/list");

                    if (!response.IsSuccessStatusCode)
                    {
                        var errorText = await response.Content.ReadAsStringAsync();
                        Console.WriteLine($"\n\u001b[31mServer response: {response.StatusCode} {errorText}\u001b[0m");
                        throw new Exception($"Failed to list keys: {response.StatusCode}");
                    }

                    var responseBody = await response.Content.ReadAsStringAsync();
                    var result = JsonSerializer.Deserialize<ListKeysResponse>(responseBody);

                    Console.WriteLine("\n\u001b[32m✅ Keys listed successfully\u001b[0m");
                    Console.WriteLine(JsonSerializer.Serialize(result, new JsonSerializerOptions 
                    { 
                        WriteIndented = true 
                    }));

                    Console.WriteLine("\nNote that neither the API key values nor the encryption fragments are included in the response.");
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error: {ex.Message}");
            }
        }
    }
}

File: java/src/main/java/com/hyperswitchai/examples/ListKeys.java
package com.hyperswitchai.examples;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class ListKeys {
    private static final String BASE_URL = "https://api.hyperswitchai.com";
    private static final ObjectMapper mapper = new ObjectMapper();
    private static final HttpClient client = HttpClient.newHttpClient();

    public static void main(String[] args) {
        try {
            String token = TokenManager.getToken();

            // List API keys
            HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(BASE_URL + "/admin/keys/list"))
                .header("Content-Type", "application/json")
                .header("Authorization", "Bearer " + token)
                .GET()
                .build();

            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

            if (response.statusCode() != 200) {
                System.out.println("\u001B[31mServer response: " + response.statusCode() + " " + response.body() + "\u001B[0m");
                throw new RuntimeException("Failed to list keys: " + response.body());
            }

            Object result = mapper.readValue(response.body(), Object.class);
            System.out.println("\u001B[32m✅ Keys listed successfully\u001B[0m");
            System.out.println(mapper.writerWithDefaultPrettyPrinter().writeValueAsString(result));
            
            System.out.println("\nNote that neither the API key values nor the encryption fragments are included in the response.");

        } catch (Exception e) {
            System.err.println("Error: " + e.getMessage());
        }
    }
} 